### PR TITLE
Update the containerd runtime binary - GCE Windows

### PR DESCRIPTION
**Pull request by ibabou**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Update the Containerd binaries to latest (1.5.2) version. The latest binary includes the fix for csi driver handling (#5411).

#### Does this PR introduce a user-facing change?

No

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 1]
```diff
@@ -1565,9 +1565,8 @@ function Install_Containerd {
   $tmp_dir = 'C:\containerd_tmp'
   New-Item $tmp_dir -ItemType 'directory' -Force | Out-Null
 
-  # TODO(ibrahimab) Change this to a gcs bucket with CI maintained and accessible by community.
```
Comment (BenTheElder) [inline cluster/gce/windows/k8s-node-setup.psm1:1568]: FWIW we host binaries for kind publicly as-is, though it is also unofficial https://github.com/kind-ci/containerd-nightlies

containerd also has amd64 binaries on their releases page

Comment (BenTheElder) [reply cluster/gce/windows/k8s-node-setup.psm1:1568]: IMO this TODO still applies, to anything in this repo.

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1568]: Yeah, I actually added the TODO when I switched it from the temp gcs bucket used earlier. The problem of keeping it only tied to the binaries available through release links, is that sometimes with certain cherry-picks needed we've to wait till it gets picked by a following release (this is the case with the csi driver issue mentioned above - that's causing failures in the gce grid). So those binaries are built from the main github containerd repo with just the cherry-pick applied and pushed similar to all other components to gke-release.

We can switch back later after releases becomes more stable, as I agree that it'll be cleaner.

Do you suggest keeping the TODO ?

Comment (BenTheElder) [reply cluster/gce/windows/k8s-node-setup.psm1:1568]: I think this is still a TODO: we can host nightly binaries on in-project resources. I also don't think we need to be running from a fork vs the latest changes in a given branch including the cherry pick there, at least on linux this works fine and we're regularly running CI not at a release commit.

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1587]: Sgtm! Kept the TODO.

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 2]
```diff
@@ -1584,9 +1584,9 @@ function Install_Containerd {
...
-  $version = '1.4.4'
-  $tar_url = ("https://github.com/containerd/containerd/releases/download/v${version}/" +
+  # TODO(ibrahimab) Change this to pick from a CI maintained and accessible by community.
+  $version = '1.5.2'
+  $tar_url = ("https://storage.googleapis.com/gke-release/winnode/containerd/${version}/" +
```
Comment (BenTheElder) [inline cluster/gce/windows/k8s-node-setup.psm1:1589]: Is this not actually 1.5.2 then based on previous discussion? because the old public download URL should work if it's not patched, and if it is patched, I think the version or some comment or _something_ should reflect that we are not actually testing unmodified 1.5.2 here.

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1589]: This is a patched one. Let me add a separate comment for that as well!

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1590]: Added!, Thanks for suggesting so Benjamin!

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 3]
```diff
@@ -1584,9 +1584,10 @@ function Install_Containerd {
   $tmp_dir = 'C:\containerd_tmp'
   New-Item $tmp_dir -ItemType 'directory' -Force | Out-Null
 
-  # TODO(ibrahimab) Change this to a gcs bucket with CI maintained and accessible by community.
-  $version = '1.4.4'
-  $tar_url = ("https://github.com/containerd/containerd/releases/download/v${version}/" +
+  # TODO(ibrahimab) Change this to pick from a CI maintained and accessible by community.
+  # Current (1.5.2) version on the gke-release bucket is patched with PR pull/5411 on containerd repo.
```
Comment (liggitt) [inline cluster/gce/windows/k8s-node-setup.psm1:1588]: I'm not in favor of changing this to unofficial versions... why is this needed here?

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1588]: I'm fine, and slightly on the side of just using the github release bins here on the main repo. The only reason is that there is csi driver issues that haven't made it yet in any release, and this causes the master containerd grid to have those tests failing (https://testgrid.k8s.io/sig-windows-gce#gce-windows-2019-containerd-master) - though, I wanted to switch to the built binaries we use on gke-release. Also, we've multiple gce/gke components that gets extracted through the prod gcs bucket, like logging agents, npd, pigz, etc.. So, not sure if going forward we need to have all changed too ?!.

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1588]: @jeremyje , any thoughts here ?

Comment (ibabou) [reply cluster/gce/windows/k8s-node-setup.psm1:1588]: I had an offline discussion with @jeremyje. Since 1.5.3 is almost there, let's wait and keep the bin pointed to the public releases then. I'll send a PR when it's available to just change the used version.
